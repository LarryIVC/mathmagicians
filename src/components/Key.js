import React from 'react';
import PropTypes from 'prop-types';
import './css/Keys.css';

const Key = ({ children, manageClick }) => {
  const isOrange = (data) => {
    let output = '';
    if (data === '+' || data === 'x' || data === '-' || data === 'รท' || data === '=') {
      output = 'orange';
    }
    return output;
  };
  return (
    <button
      className={`key ${isOrange(children)}`}
      onClick={manageClick}
      type="button"
    >
      {children}
    </button>
  );
};

Key.propTypes = {
  children: PropTypes.node.isRequired,
  manageClick: PropTypes.func.isRequired,
};

export default Key;
