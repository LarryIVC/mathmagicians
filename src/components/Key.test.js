import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Key from './Key';

describe('Key component', () => {
  it('should match the snapshot', () => {
    const { container } = render(
      <Key manageClick={() => {}}>Button Text</Key>,
    );
    expect(container).toMatchSnapshot();
  });

  it('should have the correct CSS class when data is an operator', () => {
    const { getByText } = render(
      <Key manageClick={() => {}}>+</Key>,
    );
    const button = getByText('+');
    expect(button).toHaveClass('key orange');
  });

  it('should have the default CSS class when data is not an operator', () => {
    const { getByText } = render(
      <Key manageClick={() => {}}>5</Key>,
    );
    const button = getByText('5');
    expect(button).toHaveClass('key');
  });

  it('should call manageClick when clicked', () => {
    const mockManageClick = jest.fn();
    const { getByText } = render(
      <Key manageClick={mockManageClick}>Button Text</Key>,
    );
    const button = getByText('Button Text');
    fireEvent.click(button);
    expect(mockManageClick).toHaveBeenCalled();
  });
});
