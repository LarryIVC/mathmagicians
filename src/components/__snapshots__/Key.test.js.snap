// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key component should match the snapshot 1`] = `
<div>
  <button
    class="key "
    type="button"
  >
    Button Text
  </button>
</div>
`;
